src/provider/systemdmanager/manager.rs

```diff
@@ -1,262 +1,356 @@
+                Err(anyhow::Error::from(delete_error))
+            }
         }
     }
 
-    // Write the unit file to disk and enable the service
-    // TODO: should we maybe split enable out into a different function?
-    pub fn load(
+    /// Write the proper unit file for [unit] to disk.
+    /// The location of the unit file is determined by the value of [unit_file_path]:
+    ///
+    /// * None, the unit file will be created in the base directory that this manager was initialized
+    /// with, which is either /lib/systemd/system or ~/.config/systemd/user depending on the value of
+    /// [session].
+    /// * Some<PathBuf>, the unit file will be created at this location and linked into the proper
+    /// systemd unit directory TODO: this is not implemented yet
+    ///
+    /// [force] determines if an existing unit file should be overwritten, if no  external unit file
+    /// path is specified in [unit_file_path]. If this is false and the target file exists an error
+    /// is returned.
+    ///
+    /// The value of [daemon_reload] controls whether a daemon reload is triggered after creating or
+    /// linking the unit file.
+    pub fn create_unit(
         &self,
-        name: &str,
         unit: &SystemDUnit,
-        unit_type: UnitTypes,
+        unit_file_path: Option<PathBuf>,
+        force: bool,
+        daemon_reload: bool,
     ) -> Result<(), anyhow::Error> {
         // Appends .service to name if necessary
-        let unit_name = Manager::get_unit_file_name(name, unit_type)?;
-
-        let unit_path = PathBuf::from(&unit_name);
+        let linked_unit_file = unit_file_path.is_some();
+        let unit_name = SystemdManager::get_unit_file_name(unit.name.as_ref(), &unit.unit_type)?;
 
-        // Check if an absolute path was provided
-        // if this is an absolute path the file will be written there, otherwise
-        // the file will be created in the unit directory
-        let target_file = if unit_path.is_absolute() {
-            unit_path
+        // Check if a path was provided for the unit file, otherwise use the base directory
+        let target_file = if let Some(path) = unit_file_path {
+            path.join(&unit_name)
         } else {
-            self.units_directory.join(unit_path)
+            // TODO: I think we can get away with a reference here, but not sure yet,
+            //  that would mean looking into get_unit_file_name returning a &str, _I think_
+            self.units_directory.clone().join(&unit_name)
         };
 
-        //let target_file = self.units_directory.join(&unit_name);
-        debug!("Target file for service {} : {:?}", name, target_file);
+        debug!(
+            "Target file for service [{}] : [{:?}]",
+            &unit_name, &target_file
+        );
 
+        // If no external file was created check if the file currently exists
+        if !linked_unit_file {
+            debug!("Target file [{:?}] already exists", &target_file);
```

Comment:
I think the code is correct as it is, I've added a comment to address your question above as well, which hopefully explains what I am doing..

Reply:
Actually, doing some more thinking and testing, I think the entire block should be reworked a little..

Reply:
I think I disagree.

When I see the debug statement "Target file already exists" it has not actually been checked yet if it exists.
There's nothing in that first if block other than the debug! statement.

This is what I'm proposing:

```rust
        if !linked_unit_file && target_file.exists() {
            // Target file already exists, remove if force is supplied, otherwise abort
            debug!("Target file [{:?}] already exists", &target_file);
            if force {
                self.delete_unit_file(&unit_name)?;
            } else {
                return Err(anyhow!(
                    "Target unit file [{:?}] exists and force parameter was not specified.",
                    target_file
                ));
            }
        }
```

Reply:
I have pushed a commit that should hopefully fix the entire write / delete file topic.